src/item.cpp

```diff
@@ -4132,6 +4132,10 @@ bool item::gunmod_compatible( const item& mod, bool alert ) const
     } else if( get_free_mod_locations( mod.type->gunmod->location ) <= 0 ) {
         msg = string_format( _( "Your %1$s doesn't have enough room for another %2$s mod." ), tname().c_str(), _( mod.type->gunmod->location.c_str() ) );
 
+    } else if( ammo_current() == "generic_no_ammo" ) {
+        msg = string_format( _( "You drain the charge from your %s before trying to modify it." ), tname().c_str() );
+        return true;
```

Comment:
I don't use the correct lingo, I know that. But I'm pretty sure what you said is what I said...Anyway, the return true is besides the point, it's just there to make it obvious to me when the condition was triggered.
 I need input on what ammo_current() is returning. I've seen it used in several places like this : ammo_current() != "null" , so I assumed it was returning a string containing the current ammo type, but that's apparently wrong.